p2p/session.go

```diff
@@ -216,34 +216,14 @@ func (s *session[H]) doRequest(
 	s.queue.push(stat)
 }
 
-// processResponse converts HeaderResponse to Header.
+// processResponses converts HeaderResponse to Header.
 func (s *session[H]) processResponse(responses []*p2p_pb.HeaderResponse) ([]H, error) {
-	if len(responses) == 0 {
-		return nil, errEmptyResponse
-	}
-
-	headers := make([]H, 0)
-	for _, resp := range responses {
-		err := convertStatusCodeToError(resp.StatusCode)
-		if err != nil {
-			return nil, err
-		}
-
-		var empty H
-		header := empty.New()
-		err = header.UnmarshalBinary(resp.Body)
-		if err != nil {
-			return nil, err
-		}
-		headers = append(headers, header.(H))
-	}
-
-	if len(headers) == 0 {
-		return nil, header.ErrNotFound
+	hdrs, err := processResponses[H](responses)
```

Comment:
what's the point of having this wrapper? Also the godoc doesn't match the method sig 

Also random but we should rename `s.validate` to `s.verify` because we already perform validation on response processing.

Reply:
Wdym by wrapper?

Agree on method renaming

Reply:
I mean that this method is unnecessary.

Reply:
ProcessResponses? It's a func and it's needed. See: https://github.com/celestiaorg/go-header/pull/89#discussion_r1278282010

Reply:
No we need process responses, we do not need proccessResponse.
Regardless, nit

Reply:
Ahhh, I see finally, ok

Reply:
I tried, but it does not work that nicely, both processResponses and verify have the same error-handling logic so extracting them as a separate method makes sense